### ISSUE TITLE
Routing fails when SRID of QGIS Canvas does not match SRID of edges

### ISSUE THREAD

**Issue by ArGeBre** [pgRouting/pgRoutingLayer#30]
Hi there,

in the current version of the pgRoutingLayer plugin, the routing will fail when SRID of the QGIS Canvas does not match SRID of the edges.
![grafik](https://cloud.githubusercontent.com/assets/5072877/22588899/111e0eec-ea08-11e6-9142-ed6c1d8ac65d.png)

I have edge data in EPSG:4326. When displaying a basemap using the OpenLayers plugin, the coordinate system of the QGIS canvas is switched to EPSG:3857. 

The edge data is clipped using the ST_MakeEnvelope function which is however fed with the coordinates in the display SRID:

```sql
SELECT seq, '(' || start_vid || ',' || end_vid || ')' AS path_name, path_seq AS _path_seq, 
    start_vid AS _start_vid, end_vid AS _end_vid,node AS _node, edge AS _edge, cost AS _cost,
    lead(agg_cost) over() AS _agg_cost
FROM pgr_dijkstra('
    SELECT id AS id, source AS source, target AS target, cost AS cost, 
    reverse_cost::float8 AS reverse_cost 
    FROM edge 
    WHERE edge.geom && 
    ST_MakeEnvelope( 985838.540229, 5950885.26559,1105363.9254, 6006703.79195, 4326)',
array[3436]::BIGINT[], array[3437]::BIGINT[], true)
```
Should be:
```sql
'ST_Transform(
    ST_MakeEnvelope(985838.540229, 5950885.26559, 1105363.9254, 6006703.79195, 3857),
4326)'
```

Arne

**Comment by ArGeBre** [comment]
Hi there,
has anybody been able to look at this issue? I could attempt a fix, it should be pretty straight forward, but I am not a Python man...
Arne

**Comment by cvvergara** [comment]
@ArGeBre 
I am working on the issue right now.
Will add a checkbox, to let user choose if BBOX is wanted or not

**Comment by cvvergara** [comment]
@ArGeBre 
Not downloadable from the pluging manager of QGIS yet, but if you want to give it a try, its on branch develop and just by removing the tick mark of the BBOX it wont use the canvas info.

**Comment by ArGeBre** [comment]
@cvvergara 
Good news, thanks! I'll try to check it out. I have never installed a plugin from GIT, so it's probably going to take some effort. 
I figure that just deactivating the BBOX is probably not going to fix the initial problem, though. The BBOX must be transformed from the canvas SRID to the edge layer SRID to make it work. Should be pretty straight forward to add an ST_Transform to the query when the SRIDs of canvas and egde layer are not the same.

**Comment by cvvergara** [comment]
@ArGeBre 
yes, it wont, fix that, but at least you will be able to use all the data regardless of SRID ..
btw, I merged now on master

**Comment by cayetanobv** [comment]
Fixed with this PR #47 from @veniversum
